foursight_core/react/api/yaml_utils.py

```diff
@@ -0,0 +1,57 @@
+import yaml
+
+
+def load_yaml(yaml_string: str) -> dict:
+    """
+    Loads the given YAML string and returns it as a dictionary.
+    If an error occurs on load then and empty dictionary is returned.
+    """
+    try:
+        return yaml.load(yaml_string, Loader=yaml.Loader)
+    except Exception:
+        return {}
+
+
+class _Ref(yaml.YAMLObject):
+    yaml_tag = '!Ref'
+    def __init__(self, val):
+        self.val = val
+    @classmethod
+    def from_yaml(cls, loader, node):
+        return str(node.value)
+
+
+class _Sub(yaml.YAMLObject):
+    yaml_tag = '!Sub'
+    def __init__(self, val):
+        self.val = val
+    @classmethod
+    def from_yaml(cls, loader, node):
+        return str(node.value)
+
+
+class _Join(yaml.YAMLObject):
+    yaml_tag = '!Join'
+    def __init__(self, val):
+        self.val = val
+    @classmethod
+    def from_yaml(cls, loader, node):
+        return str(node.value)
+
+
+class _GetAtt(yaml.YAMLObject):
+    yaml_tag = '!GetAtt'
+    def __init__(self, val):
+        self.val = val
+    @classmethod
+    def from_yaml(cls, loader, node):
+        return str(node.value)
+
+
+class _ImportValue(yaml.YAMLObject):
+    yaml_tag = '!ImportValue'
+    def __init__(self, val):
+        self.val = val
+    @classmethod
+    def from_yaml(cls, loader, node):
+        return str(node.value)
```

Comment:
I think all you need is something like this, yes? (I could have done this as a suggestion, but it was a bit unwieldy, so you'll have to do it manually if you agree. Needs testing, of course.)

```
class _SpecialTag(yaml.YAMLObject):
    yaml_tag = None
    def __init__(self, val):
        self.val = val
    @classmethod
    def from_yaml(cls, loader, node):
        return str(node.value)

class _Ref(_SpecialTag):
    yaml_tag = '!Ref'

class _Sub(_SpecialTag):
    yaml_tag = '!Sub'

class _Join(_SpecialTag):
    yaml_tag = '!Join'

class _GetAtt(_SpecialTag):
    yaml_tag = '!GetAtt'

class _ImportValue(_SpecialTag):
    yaml_tag = '!ImportValue'
```

Reply:
Yes good, thanks, this is much better.